App.js

```diff
@@ -306,25 +313,41 @@ export default class App extends React.Component {
             linking={linking}
             theme={navTheme}
             onStateChange={(state) => {
-              console.log('STATE CHANGED: ', state);
+              const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
+              const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
+              const isRHPOnTopOfStack = lastRHPIndex === state.index;
+              this.stateRef.current = {
+                ...this.stateRef.current,
+                state,
+                lastChatIndex,
+                lastRHPIndex,
+                isRHPOnTopOfStack
+              }
             }}
             initialState={this.state.initialState}
           >
             <Stack.Navigator
               initialRouteName="LeftHandNav"
+              screenOptions={{headersShown: false}}
             >
                 <Stack.Screen
                   name="LeftHandNav"
                   component={LeftHandNav}
-                  options={{headerShown: false}}
+                  options={{isLHNVisible: (this.stateRef.state.index === 0 && this.stateRef.routes.length === 1) || !isSmallScreenWidth}}
                 />
                 <Stack.Screen
                   name="Chat"
                   component={ChatScreen}
-                  options={{headerShown: false}}
-                  initialParams={{ id: 1 }} />
-                <Stack.Screen name="Search" component={SearchScreen} options={{headerShown: false}} />
-                <Stack.Screen name="SettingsStack" component={SettingsStackNavigator} options={{headerShown: false}} />
+                  initialParams={{id: 1}} 
+                  options={{chatIndexToRender: this.stateRef.current.lastChatIndex}}/>
```

Comment:
Same here - why is this passed to the navigator? It can be derived from the navigation state (I think).